### DISCUSSION TITLE
Using custom mask with vue-imask

### DISCUSSION THREAD

**Discussion by allard-potma** [quasarframework/quasar#12219] [General - Components / Directives / etc] [answered]
I'm trying to make an ip address input field. This is not possible with the standard Quasar masks, so I use a external mask processor, in my case [vue-imask](https://github.com/uNmAnNeR/imaskjs/tree/master/packages/vue-imask). I followed the guidelines of vue-imask and [using-third-party-mask-processors](https://quasar.dev/vue-components/input#using-third-party-mask-processorsl) , the mask works when i type something in the input, but the initial value is not loaded.
`ip_addr: ref("192.168.1.1")`
and the input:
`<q-field filled v-model="ip_addr">
<template v-slot:control="{ id, value, emitValue }">
            <input  :id="id" 
                    class="q-field__input" 
                    :model-value="value" 
                    @change="e => emitValue(e.target.value)" 
                    v-imask="ip_mask" >
        </template>
</q-field>`
and the mask object:
`ip_mask: {
                mask: 'IP.IP.IP.IP',
                blocks: {
                    IP: {
                        mask: Number,
                        scale: 0,
                        min: 0,
                        max: 255
                    }
                }
}`
With this code I expect the input to have the value 192.168.1.1 on load, but this is not the case, what's causing this? I can see ip_addr changing if I type in the input, so it seems to be working in this direction.

**Comment by MilosPaunovic** [comment]
Please provide a minimal reproduction so someone can play with it.

**Reply by allard-potma** [reply]
Created a CodeSandbox:
https://codesandbox.io/s/pedantic-jepsen-293g7?file=/src/pages/Index.vue
As you can see the normal q-input has a value at start, but the one with the mask not.

**Comment by yusufkandemir** [comment, marked as answer]
_Based on https://github.com/quasarframework/quasar/discussions/12219#discussioncomment-2058081_

Here you go: https://codesandbox.io/s/cool-goldstine-i6vj1?file=/src/pages/Index.vue

`:model-value` should be `:value` for native components, `value` in the scope should be `modelValue`.

So, [Using third party mask processors example](https://quasar.dev/vue-components/input#using-third-party-mask-processors) on the docs was wrong, I updated it on https://github.com/quasarframework/quasar/commit/965dd50002d0ec9bbe8dafd943f2347de83d15fe. Thanks for implicitly letting us know about it.
